x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameTaskFailedStateIT.java

```diff
@@ -6,87 +6,123 @@
+        createReviewsIndex();
+        String dataFrameIndex = "failure_pivot_reviews";
+        createDestinationIndexWithBadMapping(dataFrameIndex);
+        createPivotReviewsTransform(TRANSFORM_ID, dataFrameIndex, null);
+        startDataframeTransform(TRANSFORM_ID, false);
+        awaitState(TRANSFORM_ID, DataFrameTransformTaskState.FAILED);
+        Map<?, ?> fullState = getDataFrameState(TRANSFORM_ID);
+        final String failureReason = "task encountered more than 10 failures; latest failure: " +
+            "Bulk index experienced failures. See the logs of the node running the transform for details.";
+        // Verify we have failed for the expected reason
+        assertThat(XContentMapValues.extractValue("state.reason", fullState),
+            equalTo(failureReason));
+
+        // Verify that we cannot start the transform when the task is in a failed state
+        ResponseException ex = expectThrows(ResponseException.class, () -> startDataframeTransform(TRANSFORM_ID, false));
         assertThat(ex.getResponse().getStatusLine().getStatusCode(), equalTo(RestStatus.CONFLICT.getStatus()));
         assertThat(XContentMapValues.extractValue("error.reason", entityAsMap(ex.getResponse())),
             equalTo("Unable to start data frame transform [failure_pivot_1] as it is in a failed state with failure: [" +
-                "task encountered irrecoverable failure: no such index [reviews]]. " +
-                "Use force start to restart data frame transform once error is resolved."));
+                failureReason +
+                "]. Use force start to restart data frame transform once error is resolved."));
 
-        // Correct the failure by creating the reviews index again
-        createReviewsIndex();
+        // Correct the failure by deleting the destination index
+        deleteIndex(dataFrameIndex);
         // Force start the data frame to indicate failure correction
-        startDataframeTransform(transformId, true);
+        startDataframeTransform(TRANSFORM_ID, true);
         // Wait for data to be indexed appropriately and refresh for search
-        waitForDataFrameCheckpoint(transformId);
+        waitForDataFrameCheckpoint(TRANSFORM_ID);
         refreshIndex(dataFrameIndex);
 
         // Verify that we have started and that our reason is cleared
-        fullState = getDataFrameState(transformId);
+        fullState = getDataFrameState(TRANSFORM_ID);
         assertThat(XContentMapValues.extractValue("state.reason", fullState), is(nullValue()));
         assertThat(XContentMapValues.extractValue("state.task_state", fullState), equalTo("started"));
         assertThat(XContentMapValues.extractValue("state.indexer_state", fullState), equalTo("started"));
-        assertThat(XContentMapValues.extractValue("stats.search_failures", fullState), equalTo(1));
+        assertThat((int)XContentMapValues.extractValue("stats.index_failures", fullState), greaterThan(0));
```

Comment:
I purposefully made this flexible (we want to see at least ONE indexing failure) as we MAY change how we many indexing failures are allowed for specific indexing issues (e.g. mapping vs cluster issues)